v2/internal/controllers/generic_controller.go

```diff
@@ -176,7 +187,8 @@ func register(
 		RequeueDelayOverride: options.RequeueDelay,
 		PositiveConditions:   conditions.NewPositiveConditionBuilder(clock.New()),
 		//nolint:gosec // do not want cryptographic randomness here
-		rand: rand.New(lockedrand.NewSource(time.Now().UnixNano())),
+		rand:       rand.New(lockedrand.NewSource(time.Now().UnixNano())),
```

Comment:
What's the rationale for `rand` and `extensions` to be unexported when everything else on `GenericReconciler` is exported? It's a bit odd to have a mixture, particularly on an exported struct for fields that are required (I think?) - it means that some code in another package can never construct a valid instance.